com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/Project.java

```diff
@@ -772,6 +772,76 @@ private void cleanEngines(IProgress monitor, String[] platformStrings) throws IO
         m.done();
     }
 
+    private void downloadToFile(URL url, File file) throws IOException, CompileExceptionError {
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        InputStream input = null;
+        try {
+            connection.connect();
+            int code = connection.getResponseCode();
+
+            if (code == 304) {
+                logInfo("Status %d: Already cached", code);
```

Comment:
What happens if the output folder is removed between the downloads?
It's not uncommon to clean the build folder for instance (or .internal/cache)

Reply:
Not sure I understand the scenario you describe. Should we download the vanilla symbols later in the build process you mean?

Reply:
What I mean is, what would happen if you get a 304, but the target file doesn't exist? (e.g. because the build folder was removed between runs)

Reply:
Ah, got it. Yes, that's a problem.

Reply:
But we don't really handle that very well when resolving libs either.

Reply:
Now that I think of it I don't believe the server can respond with a 304 since I do not include any request headers indicating ETag or Last-Modified.

Reply:
Then I guess you shouldn't handle it here either, since if one _would_ start to send the Etag, this code would fail.